app/assets/javascripts/discourse/app/lib/hashtag-decorator.js

```diff
@@ -23,6 +25,16 @@ export function fetchUnseenHashtagsInContext(
       response[type].forEach((item) => {
         seenHashtags[type][item.ref] = seenHashtags[type][item.ref] || item;
       });
+
+      if (type === "category") {
```

Comment:
Same comment here about avoiding specific conditionals based on hardcoded type, I don't want to go backwards here with the hashtag system.

The other change I talked about could be used here too:


```javascript
hashtagTypeClass.fetchModelData(response[type].map((item) => item.id)).then((modelData) => {
  hashtagTypeClass.afterDecorate(modelData);
});